File: v1/src/test/java/com/api/v1/payment/AllPaymentsDeletionTest.java
package com.api.v1.payment;

import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class AllPaymentsDeletionTest {

    @Autowired
    private WebTestClient webTestClient;

    @Test
    @Order(1)
    void testSuccessfulAllPaymentsDeletionTest() {
        webTestClient
                .delete()
                .uri("api/v1/payments")
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

    @Test
    void testUnsuccessfulAllPaymentsDeletionTest() {
        webTestClient
                .delete()
                .uri("api/v1/payments")
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

}

File: v2/src/test/kotlin/com/api/v2/payment/AllPaymentsDeletionTest.kt
package com.api.v2.payment

import org.junit.jupiter.api.MethodOrderer
import org.junit.jupiter.api.Order
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestMethodOrder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.web.reactive.server.WebTestClient

@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
private class AllPaymentsDeletionTest {

    @Autowired
    lateinit var webTestClient: WebTestClient

    @Test
    @Order(1)
    fun testSuccessfulAllPaymentsDeletion() {
        webTestClient
            .delete()
            .uri("api/v1/payments")
            .exchange()
            .expectStatus()
            .is2xxSuccessful()
    }

    @Test
    fun testUnsuccessfulAllPaymentsDeletion() {
        webTestClient
            .delete()
            .uri("api/v1/payments")
            .exchange()
            .expectStatus()
            .is5xxServerError()
    }

}